qiime/core/type/signature.py

```diff
@@ -113,12 +114,20 @@ def decode_parameters(self, **kwargs):
         return params
 
     def check_types(self, **kwargs):
-        for key, (type_, _) in itertools.chain(self.inputs.items(),
-                                               self.parameters.items()):
+        for key, (type_, _) in self.inputs.items():
+            if kwargs[key] is None:
+                raise TypeError("Argument to input %r is None, but only "
```

Comment:
Ok, I would prefer removing it to avoid special-casing `None` where it isn't necessary.